query-connector/src/app/tests/integration/usergroup-management.test.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  getUsersWithGroupStatus,
+  addUsersToGroup,
+  removeUsersFromGroup,
+  saveUserGroupMembership,
+} from "@/app/backend/usergroup-management";
+import { getDbClient } from "@/app/backend/dbClient";
+import { User } from "@/app/models/entities/user-management";
+
+const dbClient = getDbClient();
+
+jest.mock("@/app/utils/auth", () => ({
+  adminAccessCheck: jest.fn(() => Promise.resolve(true)),
+}));
+
+const TEST_GROUP_ID = "00000000-0000-0000-0000-000000000000";
+const TEST_USER_1_ID = "00000000-0000-0000-0000-000000000001";
+const TEST_USER_2_ID = "00000000-0000-0000-0000-000000000002";
+
+describe("User Group Membership Tests", () => {
+  beforeAll(async () => {
+    await dbClient.query("BEGIN");
+
+    // Insert test users
+    const insertUsersQuery = `
+      INSERT INTO users (id, username, first_name, last_name, qc_role)
+      VALUES 
+        ($1, 'testuser1', 'Test', 'User1', 'Standard User'),
+        ($2, 'testuser2', 'Test', 'User2', 'Standard User');
+    `;
+    await dbClient.query(insertUsersQuery, [TEST_USER_1_ID, TEST_USER_2_ID]);
+
+    // Insert test group
+    const insertGroupQuery = `
+      INSERT INTO usergroup (id, name)
+      VALUES ($1, 'Test Group');
+    `;
+    await dbClient.query(insertGroupQuery, [TEST_GROUP_ID]);
+  });
+
+  afterAll(async () => {
+    try {
+      await dbClient.query(
+        "DELETE FROM usergroup_to_users WHERE usergroup_id = $1;",
+        [TEST_GROUP_ID],
+      );
+      await dbClient.query("DELETE FROM usergroup WHERE id = $1;", [
+        TEST_GROUP_ID,
+      ]);
+      await dbClient.query("DELETE FROM users WHERE id IN ($1, $2);", [
+        TEST_USER_1_ID,
+        TEST_USER_2_ID,
+      ]);
+      await dbClient.query("ROLLBACK");
+    } catch (error) {
+      console.error("Rollback failed:", error);
+    }
+  });
+
+  /**
+   * Tests retrieving user group memberships.
+   */
+  test("should retrieve user group memberships", async () => {
+    const result: User[] = await getUsersWithGroupStatus(TEST_GROUP_ID);
+
+    expect(Array.isArray(result)).toBe(true);
+    if (result.length > 0 && result[0] && typeof result[0] === "object") {
+      expect(result[0]).toHaveProperty("id");
+      expect(result[0]).toHaveProperty("username");
+      expect(result[0]).toHaveProperty("userGroupMemberships");
+
+      const membership = result[0].userGroupMemberships?.find(
+        (m) => m.usergroup_id === TEST_GROUP_ID,
+      );
+      expect(membership).toBeDefined();
+      expect(membership?.is_member).toBeDefined();
+    }
```

Comment:
```suggestion
expect(result[0]).toHaveProperty("id");
expect(result[0]).toHaveProperty("username");
expect(result[0]).toHaveProperty("userGroupMemberships");

const membership = result[0].userGroupMemberships?.find(
  (m) => m.usergroup_id === TEST_GROUP_ID,
);
expect(membership).toBeDefined();
expect(membership?.is_member).toBeDefined();
```

Think we're better off not doing this check, since in the event that `result[0]` is malformed, the if actually prevents us from catching it